internal/tracefs/kprobe.go

```diff
@@ -0,0 +1,381 @@
+package tracefs
+
+import (
+	"crypto/rand"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"syscall"
+
+	"github.com/cilium/ebpf/internal"
+	"github.com/cilium/ebpf/internal/sys"
+	"github.com/cilium/ebpf/internal/unix"
+)
+
+var (
+	ErrInvalidInput = errors.New("invalid input")
+
+	ErrInvalidMaxActive = errors.New("can only set maxactive on kretprobes")
+)
+
+func KprobeCheckLite(symbol string, pid int) error {
+	fd, err := tryAttach(symbol, pid)
```

Comment:
Why is the check here, seems like all the API is there to just move this code into `haveSyscallWrapper`?